File: tests/test-minecraft/src/main/java/pize/tests/minecraft/game/audio/MusicSet.java
package pize.tests.minecraft.game.audio;

public enum MusicSet{

    MAIN_MENU("Beginning 2", "Moog City 2", "Floating Trees", "Mutation");


    private final String[] set;

    MusicSet(String... set){
        this.set = set;
    }

    public String get(int index){
        return set[index];
    }

    public int size(){
        return set.length;
    }

}

File: tests/test-minecraft/src/main/kotlin/pize/tests/minecraft/game/audio/MusicSet.kt
package pize.tests.minecraft.game.audio

enum class MusicSet private constructor(vararg set: String) {
    MAIN_MENU("Beginning 2", "Moog City 2", "Floating Trees", "Mutation");

    private val set: Array<String>

    init {
        this.set = set as Array<String>
    }

    operator fun get(index: Int): String {
        return set[index]
    }

    fun size(): Int {
        return set.size
    }
}
